apps/nextra/components/landing/components/Carousel.tsx

```diff
@@ -0,0 +1,87 @@
+import { cva } from "class-variance-authority";
+import { Children, useEffect, useRef, useState } from "react";
+import { cn } from "utils/cn";
+
+// TODO: Currently, this component doesn't accept a value prop and therefore isn't controlled.
+// If the user is on desktop and shrinks their browser to a mobile size, they can trigger a state
+// de-sync. Once they scroll the carousel, it will be re-synced. This is an edge case so we can
+// revisit it later once we have the rest of the site done. Syncing the scroll position with a
+// value prop will likely be non-trivial to implement.
+
+export interface CarouselProps {
+  /** Each JSX.Element passed will be its own step of the carousel */
+  children: JSX.Element[];
+  /** When the current step changes, the key of the element will be given to this function */
+  onChange?: (key: string) => void;
+  /** Class name applied to the outermost wrapper of this component */
+  className?: string;
+}
+
+export function Carousel({ children, onChange, className }: CarouselProps) {
+  const [curIndex, setCurIndex] = useState(0);
+  const listRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (!onChange) return;
+    Children.forEach(children, (child, i) => {
+      if (curIndex === i && child.key !== null) onChange(child.key);
+    });
+  }, [curIndex]);
```

Comment:
I wonder if we can do this without a useEffect and useState :D 

like https://codepen.io/rsalgado/pen/EqdKjX

Reply:
I did this with `Tabs.tsx` I think you could reuse some of the code there

Reply:
We probably can. The tricky part is that the current item in the carousel is determined by the scroll position of the container element. We'll likely need a `useEffect` somewhere. I'll spend a little more time on it.